### PR TITLE
PUB-2302 - Added a11y tests to test suites

### PR DESCRIPTION
**Pull request by MindGames2015**
**Before creating a pull request make sure that:**

-   [X] commit messages are meaningful and follow good commit message guidelines
-   [X] README and other documentation has been updated / added (if needed)
-   [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/PUB-2302

### Change description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/a11y/tests/media-routes.test.ts [hunk 1]
```diff
@@ -121,5 +121,141 @@ describe('Accessibility - Media User Routes', () => {
                 testAccessibility(url, '', true, { 'search-input': 'Invalid case number' });
             });
         });
+
+        describe('Create Media Account Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:125]: Thus is a public route not a media route

File: src/test/a11y/tests/media-routes.test.ts [hunk 2]
```diff
@@ -121,5 +121,141 @@ describe('Accessibility - Media User Routes', () => {
...
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'file-upload': '' });
+            });
+
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { 'file-upload': 'Invalid File upload' });
+            });
+        });
+
+        describe('Bulk Create Media Accounts Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:161]: This is system admin route

File: src/test/a11y/tests/media-routes.test.ts [hunk 3]
```diff
@@ -121,5 +121,141 @@ describe('Accessibility - Media User Routes', () => {
...
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'bulk-account-upload': '' });
+            });
+
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { 'bulk-account-upload': 'Invalid Bulk Account Upload' });
+            });
+        });
+
+        describe('Media Account Rejection Reasons Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:173]: This is admin route

File: src/test/a11y/tests/media-routes.test.ts [hunk 4]
```diff
@@ -121,5 +121,141 @@ describe('Accessibility - Media User Routes', () => {
...
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'rejection-reasons': '' });
+            });
+
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { 'rejection-reasons': 'Invalid Check Boxes' });
+            });
+        });
+
+        describe('Bulk Create Media Accounts Confirmation Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:185]: This is system admin route

File: src/test/a11y/tests/admin-routes.test.ts [hunk 5]
```diff
@@ -83,4 +83,180 @@ describe('Accessibility - Admin Routes', () => {
             testAccessibility(route.path, route.parameter, route.postMethod, route.postBody);
         });
     });
+
+    describe('Page with Errors', () => {
+        describe('A-Z Search Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/admin-routes.test.ts:88]: This is a public route as u do not need to sign in to be able to access the page

File: src/test/a11y/tests/admin-routes.test.ts [hunk 6]
```diff
@@ -83,4 +83,180 @@ describe('Accessibility - Admin Routes', () => {
...
+            testAccessibility(url, '', true, { displayDateFrom: 'Incorrect From Date' });
+        });
+
+        describe('with invalid input data', () => {
+            testAccessibility(url, '', true, { displayDateTo: 'Incorrect To Date' });
+        });
+    });
+
+    describe('Manual Upload Summary Page', () => {
+        describe('A-Z Search Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/admin-routes.test.ts:186]: why is there is 'describe' for A-Z search page but u r testing manual-upload-summary

File: src/test/a11y/tests/admin-routes.test.ts [hunk 7]
```diff
@@ -83,4 +83,180 @@ describe('Accessibility - Admin Routes', () => {
...
+                testAccessibility(url, '', true, { 'manual-upload': '' });
+            });
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'view-choice': 'Invalid Choice' });
+            });
+        });
+    });
+
+    describe('User Management Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/admin-routes.test.ts:199]: This is a system admin page

File: src/test/a11y/tests/admin-routes.test.ts (hunk 5 above)
Comment (MindGames2015) [reply src/test/a11y/tests/admin-routes.test.ts:88]: Modified test. Thanks Kian.

File: src/test/a11y/tests/admin-routes.test.ts (hunk 7 above)
Comment (MindGames2015) [reply src/test/a11y/tests/admin-routes.test.ts:199]: Moved test to system admin page. Thanks Kian.

File: src/test/a11y/tests/media-routes.test.ts (hunk 3 above)
Comment (MindGames2015) [reply src/test/a11y/tests/media-routes.test.ts:173]: Moved test to admin route. Thanks Kian.

File: src/test/a11y/tests/media-routes.test.ts (hunk 2 above)
Comment (MindGames2015) [reply src/test/a11y/tests/media-routes.test.ts:161]: Moved test to admin route. Thanks Kian

File: src/test/a11y/tests/media-routes.test.ts (hunk 1 above)
Comment (MindGames2015) [reply src/test/a11y/tests/media-routes.test.ts:125]: Moved test to public route. Thanks Kian.

**Review by KianKwa — CHANGES_REQUESTED** [review]
A lot of the new tests are rendering exactly the same error page. There is no need for test duplication. It is just slowing down the build time with no benefit.

File: src/test/a11y/tests/admin-routes.test.ts [hunk 8]
```diff
@@ -84,4 +84,42 @@ describe('Accessibility - Admin Routes', () => {
             testAccessibility(route.path, route.parameter, route.postMethod, route.postBody);
         });
     });
+
+    describe('Page with Errors', () => {
+        describe('Create Admin Account Page', () => {
+            const url = '/create-admin-account';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/admin-routes.test.ts:93]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/admin-routes.test.ts [hunk 9]
```diff
@@ -84,4 +84,42 @@ describe('Accessibility - Admin Routes', () => {
...
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { listType: 'EMPTY' });
+            });
+        });
+
+        describe('Remove List Search Page', () => {
+            const url = '/remove-list-search';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'search-input': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/admin-routes.test.ts:121]: 'search-input' is not the right post body for this test.

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 10]
```diff
@@ -125,4 +125,46 @@ describe('Accessibility - System Admin Routes', () => {
...
+
+        describe('Bulk Create Media Accounts Page', () => {
+            const url = '/bulk-create-media-accounts';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
+            });
+        });
+
+        describe('User Management Page', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:162]: This route will never throw any error. Remove this test

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 11]
```diff
@@ -125,4 +125,46 @@ describe('Accessibility - System Admin Routes', () => {
             testAccessibility(route.path, route.parameter);
         });
     });
+
+    describe('Page with Errors', () => {
+        describe('Create System Admin Account Page', () => {
+            const url = '/create-system-admin-account';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:134]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 12]
```diff
@@ -125,4 +125,46 @@ describe('Accessibility - System Admin Routes', () => {
...
+            });
+        });
+
+        describe('Reference Data upload Page', () => {
+            const url = '/reference-data-upload';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:142]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 13]
```diff
@@ -125,4 +125,46 @@ describe('Accessibility - System Admin Routes', () => {
...
+            });
+        });
+
+        describe('Delete Court Reference Data Confirmation Page', () => {
+            const url = '/delete-court-reference-data-confirmation';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:150]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 14]
```diff
@@ -125,4 +125,46 @@ describe('Accessibility - System Admin Routes', () => {
...
+            });
+        });
+
+        describe('Bulk Create Media Accounts Page', () => {
+            const url = '/bulk-create-media-accounts';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:158]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/public-routes.test.ts [hunk 15]
```diff
@@ -44,6 +44,21 @@ describe('Accessibility - Public Routes', () => {
...
             describe('with no input data', () => {
                 const url = '/search';
+                testAccessibility(url, '', true, { 'search-input': '' });
+            });
+        });
+
+        describe('Sign-in Page', () => {
+            describe('with no input data', () => {
+                const url = '/sign-in';
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/public-routes.test.ts:54]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/public-routes.test.ts [hunk 16]
```diff
@@ -44,6 +44,21 @@ describe('Accessibility - Public Routes', () => {
...
+            });
+        });
+
+        describe('View option  Page', () => {
+            const url = '/view-option';
+
+            describe('with no input data', () => {
                 testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/public-routes.test.ts:62]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/media-routes.test.ts [hunk 17]
```diff
@@ -122,5 +122,39 @@ describe('Accessibility - Media User Routes', () => {
...
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { 'search-input': 'Invalid case number' });
+            });
+        });
+
+        describe('Subscription Add Page', () => {
+            const url = '/subscription-add';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:156]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/media-routes.test.ts [hunk 18]
```diff
@@ -122,5 +122,39 @@ describe('Accessibility - Media User Routes', () => {
                 testAccessibility(url, '', true, { 'search-input': 'Invalid case number' });
             });
         });
+
+        describe('Bulk Unsubscribe Confirmation Page', () => {
+            const url = '/bulk-unsubscribe-confirmation';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'input-autocomplete': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:130]: 'input-autocomplete' is not the right post body for this test. Please check the controller code

File: src/test/a11y/tests/media-routes.test.ts [hunk 19]
```diff
@@ -122,5 +122,39 @@ describe('Accessibility - Media User Routes', () => {
...
+            const url = '/pending-subscriptions';
+
+            beforeEach(() => {
+                sinon.restore();
+                sinon.stub(PublicationService.prototype, 'getCaseByCaseNumber').resolves(null);
+                sinon.stub(PublicationService.prototype, 'getCaseByCaseUrn').resolves(null);
+            });
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'search-input': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:144]: 'search-input' is not the right post body for this test.

File: src/test/a11y/tests/media-routes.test.ts [hunk 20]
```diff
@@ -122,5 +122,39 @@ describe('Accessibility - Media User Routes', () => {
...
+            });
+
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, { 'search-input': 'Invalid case number' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:148]: 'search-input' is not the right post body for this test.

File: src/test/a11y/tests/system-admin-routes.test.ts (hunk 10 above)
Comment (MindGames2015) [reply src/test/a11y/tests/system-admin-routes.test.ts:162]: Removed test. Thanks Kian

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 21]
```diff
@@ -125,4 +125,38 @@ describe('Accessibility - System Admin Routes', () => {
             testAccessibility(route.path, route.parameter);
         });
     });
+
+    describe('Page with Errors', () => {
+        describe('Create System Admin Account Page', () => {
+            const url = '/create-system-admin-account';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, {'create-system-admin-account': ''});
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:134]: 'create-system-admin-account' is not the right postBody. Check the controller

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 22]
```diff
@@ -125,4 +125,38 @@ describe('Accessibility - System Admin Routes', () => {
...
+            });
+        });
+
+        describe('Reference Data upload Page', () => {
+            const url = '/reference-data-upload';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, {});
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:142]: {} is the default value for testAccessibility(). No need to pass in the default value

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 23]
```diff
@@ -125,4 +125,38 @@ describe('Accessibility - System Admin Routes', () => {
...
+            describe('with invalid input data', () => {
+                testAccessibility(url, '', true, {'delete-choice': null});
+            });
+        });
+
+        describe('Bulk Create Media Accounts Page', () => {
+            const url = '/bulk-create-media-accounts';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, {});
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:158]: {} is the default value for testAccessibility(). No need to pass in the default value

File: src/test/a11y/tests/media-routes.test.ts [hunk 24]
```diff
@@ -111,15 +111,34 @@ describe('Accessibility - Media User Routes', () => {
             beforeEach(() => {
                 sinon.restore();
                 sinon.stub(PublicationService.prototype, 'getCaseByCaseNumber').resolves(null);
-                sinon.stub(PublicationService.prototype, 'getCaseByCaseUrn').resolves(null);
             });
 
             describe('with no input data', () => {
-                testAccessibility(url, '', true, { 'search-input': '' });
+                testAccessibility(url, '', true, { 'case-number': '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:117]: Why are you changing the existing code? It was right before

File: src/test/a11y/tests/media-routes.test.ts [hunk 25]
```diff
@@ -111,15 +111,34 @@ describe('Accessibility - Media User Routes', () => {
             beforeEach(() => {
                 sinon.restore();
                 sinon.stub(PublicationService.prototype, 'getCaseByCaseNumber').resolves(null);
-                sinon.stub(PublicationService.prototype, 'getCaseByCaseUrn').resolves(null);
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:114]: This is needed for the existing test to work

File: src/test/a11y/tests/media-routes.test.ts [hunk 26]
```diff
@@ -111,15 +111,34 @@ describe('Accessibility - Media User Routes', () => {
...
             });
 
             describe('with no input data', () => {
-                testAccessibility(url, '', true, { 'search-input': '' });
+                testAccessibility(url, '', true, { 'case-number': '' });
             });
+        });
 
-            describe('with invalid input data', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:121]: Why is the existing test got removed?

File: src/test/a11y/tests/media-routes.test.ts [hunk 27]
```diff
@@ -111,15 +111,34 @@ describe('Accessibility - Media User Routes', () => {
...
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { 'bulk-unsubscribe-choice': '' });
+            });
+        });
+
+        describe('Pending Subscriptions Page', () => {
+            const url = '/pending-subscriptions';
+
+            describe('with no input data', () => {
+                testAccessibility(url, '', true, { userId: '' });
```
Comment (KianKwa) [inline src/test/a11y/tests/media-routes.test.ts:133]: userId is not the right postBody, it is part of req.user. In fact this route does not have a postBody so no need to pass in anything

File: src/test/a11y/tests/system-admin-routes.test.ts [hunk 28]
```diff
@@ -125,4 +125,38 @@ describe('Accessibility - System Admin Routes', () => {
...
+            });
+        });
+
+        describe('Delete Court Reference Data Confirmation Page', () => {
+            const url = '/delete-court-reference-data-confirmation';
+
+            describe('with invalid input data', () => {
```
Comment (KianKwa) [inline src/test/a11y/tests/system-admin-routes.test.ts:149]: Description should be 'with no input data' as there is no data passing in

**Review by KianKwa — APPROVED** [review]

**Pull request merged**